controllers/articles.js

```diff
@@ -239,6 +239,149 @@ const articles = {
       }
     }
   },
+
+  /*
+  * Bookmarking and article so that one can come back
+  * and read it later.
+  */
+  createBookmark: async (req, res) => {
+    try {
+      const { slug } = req.params;
+      const { id: userId } = req.user;
+      if (!userId) {
+        res.status(400).send({
+          status: res.statusCode,
+          errorMessage: 'Please first login to bookmark this article',
+        });
+      }
+      const findArticle = await article.findOne({ where: { slug } });
+      const findUser = await User.findOne({ where: { id: findArticle.author } });
+      const checkExist = await bookmark.findOne({ where: { articleId: findArticle.id, userId } });
+      if (checkExist) {
+        return res.status(400).send({
+          status: res.statusCode,
+          errorMessage: 'You have already bookmarked this article',
+        });
+      }
+
+      const createBookmark = await bookmark.create({
+        userId, articleId: findArticle.id
+      });
+      if (createBookmark.articleId) {
+        return res.status(200).send({
+          status: res.statusCode,
+          message: `Article from ${findUser.username} has been bookmarked`,
+          data: {
+            title: findArticle.title,
+            body: findArticle.body,
+            description: findArticle.description,
+          }
+        });
+      }
+      return res.status(404).send({
+        status: res.statusCode,
+        errorMessage: 'The article your trying to bookmark does not exit',
+      });
+    } catch (error) {
+      if (error.message) {
+        return res.status(400).send({
+          status: res.statusCode,
+          errorMessage: 'Something went wrong, artilce does not exits anymore',
```

Comment:
Most of the errors that come in the catch block are not caused by the client side. It's better if you change to 500 because most of them cause by the server side.